Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+    {
+        Write-Host "API Management services details have been backed up to $($backupFile)" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "HTTPS url scheme will be enabled on the API(s) of all API Management services." -ForegroundColor $([Constants]::MessageType.Warning)
+
+        if (-not $Force)
+        {
+            Write-Host "Do you want to enable HTTPS url scheme on the API(s) of all API Management services? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "HTTPS url scheme will not be enabled for any API(s) of API Management Service. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+                break
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. HTTPS url scheme will be enabled on the API(s) of all API Management services without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Enabling HTTPS url scheme for API(s) of API Management services..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+        # To hold results from the remediation.
+        $apiManagementServicesRemediated = @()
+        $apiManagementServicesSkipped = @()
+
+        $apiManagementWithoutHttpsEnabled | ForEach-Object {
+            $apiManagementservice = $_
+            $resourceGroupName = $_.ResourceGroupName
+           
+            Write-Host "Enabling HTTPS url scheme for API(s) of API Management Service: Resource ID - $($_.ResourceId), Resource Group Name - $($_.ResourceGroupName), Resource Name - $($_.ResourceName)" -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            try
+            {   #To hold name of API(s) which are remediated and skipped.        
+                $listOfApiRemediated = @()
+                $listOfApiSkipped = @()
+                
+                $apiMgmtContext = New-AzAPIManagementContext -ResourceGroupName $_.ResourceGroupName -ServiceName $_.ResourceName
+                $apiMgmt = Get-AzAPIManagementAPI -Context $apiMgmtContext
+                $apiManagementservice.ListOfAPIWithDisabledHttps = $apiManagementservice.ListOfAPIWithDisabledHttps -split ","
+                $apiMgmt | ForEach-Object { 
+                    if($apiManagementservice.ListOfAPIWithDisabledHttps -contains  $_.Name)
+                    {
+                        Set-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId -Protocols @("Https") -ErrorAction SilentlyContinue
+                        $output = (Get-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId  -ErrorAction SilentlyContinue).Protocols 
+                   
+                        if($output -eq "Https")
+                        {
+                       
```

Comment:
Please delete the stray newline.
`$output` will be a list of protocols. Please confirm this.